### ISSUE TITLE
DALLE trained on FashionGen Dataset RESULTS 💯 

### ISSUE THREAD

**Issue by alexriedel1** [lucidrains/DALLE-pytorch#443]
### DALLE on FashionGen

- I trained **Dall-E + VQGAN** on the **FashionGen** dataset (https://arxiv.org/abs/1806.08317) on **Google Colab** and got decent results. 
- Without the VQGAN training on the FashionGen dataset, DALLE is really bad at generating faces which makes clothing generations looking extremely strange.

### Text to image generation and re-ranking by CLIP
Best 16 from 48 generations ranked by CLIP
### Generations from the training set (Including their Groundtruths)

![Download (4)](https://user-images.githubusercontent.com/54716527/187938574-ac5f985b-242d-41c5-a92b-5cd688436036.jpg)
![Download (5)](https://user-images.githubusercontent.com/54716527/187938577-f98de296-a176-4d3d-a233-366eea8902d8.jpg)
![Download (6)](https://user-images.githubusercontent.com/54716527/187938578-d33a1f65-1b92-4111-872d-67b0b2155c13.jpg)
![Download (7)](https://user-images.githubusercontent.com/54716527/187938579-926055fa-f1d5-4a23-8b1c-a5864e14ded5.jpg)
![Download (8)](https://user-images.githubusercontent.com/54716527/187938582-88f329a1-e115-4fbc-a6aa-8b813036b45d.jpg)

### Generations based on custom prompts (withouttheir Groundtruths)
![Download (1)](https://user-images.githubusercontent.com/54716527/187938873-9b190a17-f518-41c1-b467-c4d50b86b657.jpg)
![Download (2)](https://user-images.githubusercontent.com/54716527/187938878-799f6416-5148-4ce8-901f-c4018f56ddb6.jpg)
![Download (3)](https://user-images.githubusercontent.com/54716527/187938880-6021c6c3-d6ed-4718-92c5-41916401d86d.jpg)
![Download (9)](https://user-images.githubusercontent.com/54716527/187938881-a74d5e4d-09e8-43f7-850a-238ee864e499.jpg)
![Download](https://user-images.githubusercontent.com/54716527/187938886-32b81e9a-60dd-4241-a274-f66f04527528.jpg)

### Model specifications
**VAE**
Trained VQGAN for **1 epoch** on Fashion-Gen dataset
Embeddings: 1024
Batch size: 5

**DALLE**
Trained DALLE for **1 epoch** on  Fashion-Gen dataset
dim = 312
text_seq_len = 80
depth = 36
heads = 12
dim_head = 64
reversible = 0
attn_types =('full', 'axial_row', 'axial_col', 'conv_like')

**Optimization**
Optimizer: Adam
Learning rate: 4.5e-4
Gradient Clipping: 0.5
Batch size: 7

![image](https://user-images.githubusercontent.com/54716527/187939257-fa941858-853d-463a-97a2-0311b8418915.png)

**Comment by Asthestarsfalll** [comment]
Hi, can you offer the Colab link and check points?

**Comment by alexriedel1** [comment]
> Hi, can you offer the Colab link and check points?

You'll find the trained Dall-E weights here:  https://drive.google.com/uc?id=1kEHTTZH2YbbHZjY6fTWuPb5_D-7nQ866

**Comment by Asthestarsfalll** [comment]
@alexriedel1 
Thank you! 
And I'm wondering which vocab you use, I only have the bpe_simple_vocab_16e6 supplied by openai

**Comment by Asthestarsfalll** [comment]
I download the weights, but it seems that it's parameters are different.
![image](https://user-images.githubusercontent.com/72954905/198887549-6884e714-e66e-4d96-af0a-3bce2fecfc72.png)

**Comment by alexriedel1** [comment]
Yes right, the text sequence length is 120, is this a problem for you?

**Comment by Asthestarsfalll** [comment]
No, It' s just different from the description of the model.
![image](https://user-images.githubusercontent.com/72954905/199211516-fe16e507-0132-4edb-8ad8-439b7feba2ba.png)
I'm wondering which bpe file you use, and why the num_text_tokens are such long.

**Comment by Asthestarsfalll** [comment]
Thank you a lot!

**Comment by killah-t-cell** [comment]
Hi do you still have access to the Fashiongen dataset? I can't seem to find a good link for it.
